go/test/endtoend/vtgate/queries/kill/kill_test.go

```diff
@@ -0,0 +1,234 @@
+	require.ErrorContains(t, err, "EOF (errno 2013) (sqlstate HY000)")
+}
+
+// TestKillDifferentConnection kills different connection and check relevant error messages.
+func TestKillDifferentConnection(t *testing.T) {
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	killConn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer killConn.Close()
+
+	// connection does not exist
+	_, err = utils.ExecAllowError(t, killConn, "kill 12345")
+	require.ErrorContains(t, err, "Unknown thread id: 12345 (errno 1094) (sqlstate HY000)")
+
+	// connection exist
+	_, err = utils.ExecAllowError(t, killConn, fmt.Sprintf("kill %d", conn.ConnectionID))
+	require.NoError(t, err)
+
+	// executing on closed connection
+	_, err = utils.ExecAllowError(t, conn, "select 1")
+	require.ErrorContains(t, err, "EOF (errno 2013) (sqlstate HY000)")
+}
+
+// TestKillOwnQuery kills the kill statement itself
+func TestKillOwnQuery(t *testing.T) {
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	_, err = utils.ExecAllowError(t, conn, fmt.Sprintf("kill query %d", conn.ConnectionID))
+	// TODO: does not really change anything, but expect to receive Queery Interrupted error
+	//  "(errno 1317) (sqlstate 70100)"
+	require.NoError(t, err)
+}
+
+// TestKillDifferentConnectionQuery kills query on different connection and check relevant error messages.
+func TestKillDifferentConnectionQuery(t *testing.T) {
+	setupData(t, false)
+	defer dropData(t)
+
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	killConn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer killConn.Close()
+
+	// killing query on non-existent connection
+	_, err = utils.ExecAllowError(t, killConn, "kill query 12345")
+	require.ErrorContains(t, err, "Unknown thread id: 12345 (errno 1094) (sqlstate HY000)")
+
+	done := make(chan error)
+	go func() {
+		// 20 seconds sleep. Should be stopped by kill statement.
+		_, err := utils.ExecAllowError(t, conn, "select sleep(20) from test")
+		done <- err
+	}()
+
+	for {
+		select {
+		case execErr := <-done:
+			require.ErrorContains(t, execErr, "context canceled (errno 1317) (sqlstate 70100)")
+			return
+		case <-time.After(100 * time.Millisecond):
+			_, err = utils.ExecAllowError(t, killConn, fmt.Sprintf("kill query %d", conn.ConnectionID))
+			require.NoError(t, err)
+		case <-time.After(5 * time.Second):
+			t.Fatal("test did not complete in 5 seconds.")
+		}
+	}
+}
+
+// TestKillOnHungQuery test that any hung query should return.
+func TestKillOnHungQuery(t *testing.T) {
+
+	t.Run("connection close", func(t *testing.T) {
+		testHungQuery(t, func(conn1 *mysql.Conn, conn2 *mysql.Conn) {
+			// closing the hung query connection.
+			conn2.Close()
+		}, "(errno 2013) (sqlstate HY000)")
```

Comment:
not following why this is the expected result here

Reply:
This is the expected error code for `Lost connection to MySQL server during query` as the connection is closed on which the response is expected.